OSDOCS#17844: Document replacing a healthy etcd member (#112025)
openshift/openshift-docs: modules/determining-replace-healthy-etcd-member.adoc

// Module included in the following assemblies:
//
// * backup_and_restore/control_plane_backup_and_restore/replacing-healthy-etcd-member.adoc
// * etcd/etcd-backup-restore/replace-healthy-etcd-member.adoc

:_mod-docs-content-type: PROCEDURE
[id="determining-replace-healthy-etcd-member_{context}"]
= Determining how to replace a healthy etcd member

[role="_abstract"]
To choose the correct procedure for replacing a healthy etcd member, check whether your cluster uses the {ai-full}, a control plane machine set, or the Machine API. Use the {oc-first} to identify your cluster configuration and follow the matching replacement procedure.

.Prerequisites

> 

Reviewer: The SSG says to avoid using imperative forms in Prerequisites sections. https://redhat-documentation.github.io/supplementary-style-guide/#prerequisites
```
* You installed the {oc-first}.
* You logged in to `oc` as a user with the `cluster-admin` role.
```

* Install the {oc-first}.
* Log in to `oc` as a user with the `cluster-admin` role.

.Procedure

. Check whether the cluster was installed by using the {ai-full} by running the following command:
+
[source,terminal]
----
$ oc get agentclusterinstall -A
----
+

> * If the command returns one or more `AgentClusterInstall` resources, follow the procedure to replace a control plane node in a healthy cluster in the {ai-full} documentation in Additional resources.

Reviewer: ```suggestion
* If the command returns one or more `AgentClusterInstall` resources, follow the procedure in "Replacing a control plane node in a healthy cluster" in the {ai-full} documentation.
```

+
* If the command returns no resources, continue with the following steps.

. Check whether the cluster has a control plane machine set by running the following command:
+
[source,terminal]
----
$ oc -n openshift-machine-api get controlplanemachineset
----
+

> * If the command returns a `ControlPlaneMachineSet` resource, follow "Replacing a healthy etcd member with a control plane machine set".

Reviewer: ```suggestion
* If the command returns a `ControlPlaneMachineSet` resource, follow the procedure in "Replacing a healthy etcd member with a control plane machine set".
```

+
* If the command returns no resources, continue to the next step.

> . Check if the cluster has control plane `Machine` objects by running the following command:

Reviewer: ```suggestion
. Check whether the cluster has control plane `Machine` objects by running the following command:
```
Tiny edit to be consistent with wording in other steps.

+
[source,terminal]
----
$ oc get machines -l machine.openshift.io/cluster-api-machine-role=master -n openshift-machine-api
----
+

> * If `Machine` objects exist, follow "Replacing a healthy etcd member with the Machine API".

Reviewer: ```suggestion
* If `Machine` objects exist, follow the procedure in "Replacing a healthy etcd member with the Machine API".
```

+

> * If there are no `Machine` objects, follow "Replacing a healthy etcd member by scaling up and scaling down".

Reviewer: ```suggestion
* If there are no `Machine` objects, follow the procedure in "Replacing a healthy etcd member by scaling up and scaling down".
```
